numba/cuda/tests/cudapy/test_intrinsics.py

```diff
@@ -440,6 +440,30 @@ def test_simple_gridsize1d(self):
         compiled[nctaid, ntid](ary)
         self.assertEqual(ary[0], nctaid * ntid)
 
+    @skip_on_cudasim('Requires too many threads')
+    def test_issue_9229(self):
+        # Ensure that grid and grid size are correct - #9229 showed that they
+        # overflowed an int32.
+        @cuda.jit
+        def f(grid_error, gridsize_error):
+            i1 = cuda.grid(1)
+            i2 = cuda.blockIdx.x * cuda.blockDim.x + cuda.threadIdx.x
+            gs1 = cuda.gridsize(1)
+            gs2 = cuda.blockDim.x * cuda.gridDim.x
+            if i1 != i2:
+                grid_error[0] = 1
+            if gs1 != gs2:
+                gridsize_error[0] = 1
+
+        grid_error = np.zeros(1, dtype=np.uint64)
+        gridsize_error = np.zeros(1, dtype=np.uint64)
+
+        # A lerge enough grid for thread IDs to overflow an int32
+        f[22121216, 256](grid_error, gridsize_error)
```

Comment:
Should the blocks per grid be more easily related to the size of an int32 given TPB is 256?

Reply:
I felt the way I'd written it (taken in conjunction with the comment on the previous line) was sufficient - if you'd like a specific change could you please suggest what you have in mind?

Reply:
I'd guess at `(2 ** 32 // 256) + 1` but a comment suggesting the source of the current number would be ok too (I couldn't work out why that specific number was there, but realise now it's from the reproducer in the ticket). Thanks!